c/tskit/trees.h

```diff
@@ -111,6 +111,37 @@ typedef struct {
     tsk_table_collection_t *tables;
 } tsk_treeseq_t;
 
+typedef struct {
+    tsk_id_t index;
+    struct {
+        double left;
+        double right;
+    } interval;
+    struct {
+        tsk_id_t start;
+        tsk_id_t stop;
+        const tsk_id_t *order;
+    } in;
+    struct {
+        tsk_id_t start;
+        tsk_id_t stop;
+        const tsk_id_t *order;
+    } out;
+    tsk_id_t left_current_index;
+    tsk_id_t right_current_index;
+    int direction;
+    const tsk_treeseq_t *tree_sequence;
+} tsk_tree_position_t;
+
+int tsk_tree_position_init(
```

Comment:
Methods are declared further down the file, separate from struct definitions. Just move the function signatures to the bottom of the file.